components/Banner/examples/BannerTypicalUseExampleViewController.m

```diff
@@ -129,230 +130,238 @@ + (NSDictionary *)catalogMetadata {
+                            exampleUseSelector:@selector(showMultiLineAlignedButtonStyleBanner)];
   [bannerExampleList addObject:exampleUseInfo3];
 
+  BannerExampleUseInfo *exampleUseInfo4 = [BannerExampleUseInfo
+      infoWithIdentifier:@"example4"
+             displayName:@"Long Text with Two Short Actions (Icon)"
+        exampleUseTarget:self
+      exampleUseSelector:@selector(showMultiLineAlignedButtonStyleBannerWithIcon)];
+  [bannerExampleList addObject:exampleUseInfo4];
+
+  BannerExampleUseInfo *exampleUseInfo5 =
+      [BannerExampleUseInfo infoWithIdentifier:@"example5"
+                                   displayName:@"Long Text with Two Long Actions"
+                              exampleUseTarget:self
+                            exampleUseSelector:@selector(showMultiLineStackedButtonStyleBanner)];
+  [bannerExampleList addObject:exampleUseInfo5];
+
+  BannerExampleUseInfo *exampleUseInfo6 = [BannerExampleUseInfo
+      infoWithIdentifier:@"example6"
+             displayName:@"Long Text with Two Long Actions (Icon)"
+        exampleUseTarget:self
+      exampleUseSelector:@selector(showMultiLineStackedButtonStyleBannerWithIcon)];
+  [bannerExampleList addObject:exampleUseInfo6];
+
   return [bannerExampleList copy];
 }
 
 #pragma mark - Example handlers
 
 - (void)showSingleLineStyleBanner {
-  if (self.bannerViewContainer) {
-    [self.bannerViewContainer removeFromSuperview];
+  if (self.bannerView) {
+    [self.bannerView removeFromSuperview];
   }
-  // Set up banner view container
-  UIView *bannerViewContainer = [[UIView alloc] initWithFrame:CGRectZero];
-  [self.view addSubview:bannerViewContainer];
-  bannerViewContainer.backgroundColor = [UIColor whiteColor];
-  self.bannerViewContainer = bannerViewContainer;
 
   MDCBannerView *bannerView = [[MDCBannerView alloc] init];
-  bannerView.image = [UIImage imageNamed:@"Email"];
-  bannerView.text = exampleShortText;
-
-  MDCButton *button = [bannerView.buttons firstObject];
-  [button setTitle:@"DISMISS" forState:UIControlStateNormal];
-  UIFont *buttonFont = [MDCTypography body2Font];
-  [button setTitleFont:buttonFont forState:UIControlStateNormal];
-  [button setTitleFont:buttonFont forState:UIControlStateHighlighted];
+  bannerView.textLabel.text = exampleShortText;
+  bannerView.backgroundColor = self.colorScheme.surfaceColor;
+  UIEdgeInsets margins = UIEdgeInsetsZero;
+  margins.left = exampleBannerContentPadding;
+  margins.right = exampleBannerContentPadding;
+  bannerView.layoutMargins = margins;
+  [self.view addSubview:bannerView];
+  self.bannerView = bannerView;
+
+  MDCButton *button = bannerView.leadingButton;
+  [button setTitle:@"Dismiss" forState:UIControlStateNormal];
+  button.uppercaseTitle = YES;
   [button setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
   button.backgroundColor = self.colorScheme.surfaceColor;
+  bannerView.trailingButton.hidden = YES;
```

Comment:
This is somewhat surprising: is it expected that clients must hide the buttons they don't want to be shown?

I see two paths here:

1. Buttons are not hidden by default. We need to document that clients are expected to hide buttons that should not be shown.
2. Buttons are hidden by default. We need to document that clients are expected to unhide buttons if they want them to appear.

I'm not certain that either of these realities are particularly great, but either way we need to document this expected interaction with the component's buttons (not in this PR, but in a separate PR to the MDCBannerView public API).

Reply:
I definitely agree. It is one of the few parts I feel unclear on the API, which is too implicit for developer to use. I will document it in the public API. https://github.com/material-components/material-components-ios/issues/7001 created to track the follow up work.

Reply:
@wenyuzhang666 I thought we spoke and these properties would function similarly to UIButton. If they were never accessed, they would not be created. If they were accessed, they would be created dynamically.  This line looks like it's purpose is the side-effect of creating the button.

Reply:
@romoore I remember the conversation. I double checked the behavior of `UIButton` after that,  UIButton's `imageView` is always created and set hidden by default if no image is set, which led to the non-lazy load approach right now. I can go with the lazy load one too.